src/train-unet-from-scratch.py

```diff
@@ -63,33 +54,19 @@
 logger = logging.getLogger(__name__)
 
 
-def image_grid(imgs, rows, cols):
-    assert len(imgs) == rows * cols
-
-    w, h = imgs[0].size
-    grid = Image.new("RGB", size=(cols * w, rows * h))
-    grid_w, grid_h = grid.size
-
-    for i, img in enumerate(imgs):
-        grid.paste(img, box=(i % cols * w, i // cols * h))
-    return grid
-
-
-def log_validation(pipeline, pipeline_params, tokenizer, args, rng, weight_dtype):
+def generate_images(pipeline, pipeline_params, prompts, rng, report_to: str = ""):
     logger.info("Running validation...")
 
+    # TODO(bruno): Why do we need to copy the params?
     pipeline_params = pipeline_params.copy()
 
     num_samples = jax.device_count()
     prng_seed = jax.random.split(rng, jax.device_count())
 
-    validation_prompts = args.validation_prompt
-    
     image_logs = []
-
-    for validation_prompt in validation_prompts:
-        prompts = num_samples * [validation_prompt]
-        prompt_ids = pipeline.prepare_text_inputs(prompts)
+    for prompt in prompts:
+        ps = num_samples * [prompt]
+        prompt_ids = pipeline.prepare_inputs(ps)
         prompt_ids = shard(prompt_ids)
```

Comment:
I'm not sure I follow this change. Are prompts the current prompts being used for training? Because we want the validation prompt to not change right?

Reply:
The validation prompts are still being used. I just cleaned up the code a bit.